dask_match/core.py

```diff
@@ -524,6 +590,9 @@ def _replacement_rules(cls):
 
         yield from super()._replacement_rules()
 
+        # Push down partition projection
+        yield partition_pushdown_rule(cls)
```

Comment:
Could this be done in a superclass somehow, like `Blockwise`?

Reply:
Also, with the current rule as written, does the head/tail operation get pushed down into each side of the `Add/Sub/...`